tsdb/engine/tsm1/reader.go

```diff
@@ -607,7 +607,7 @@ func (d *indirectIndex) UnmarshalBinary(b []byte) error {
 		i++
 
 		// 2 byte count of index entries
-		count := btou16(b[i : i+indexCountSize])
+		count := int32(btou16(b[i : i+indexCountSize]))
```

Comment:
I.e. by its nature `btou16` returns a 16-bit type. And since `count` is never incremented, how it could it overflow?